### PR TITLE
NCBC-3892: Phase 4 eager loading — nested owned-type materialisation and post-merge fixes

### PR DESCRIPTION
**Pull request by jeffrymorris**
Completes Phase 4 of the Couchbase EF Core eager loading implementation.

Extracted a recursive MaterializeOwnedItem helper to correctly read OwnsOne/OwnsMany at arbitrary nesting depth from embedded JSON
Fixed SQL generation to skip lateral-join subqueries EF Core emits for nested owned navigations
Fixed the write-path ownership-chain walk to traverse all levels, preventing "Invalid keyspace format" errors on nested owned types
Added SerializeOwnedItem recursive helper for nested owned write serialisation
Fixed MaterializeOwnedItem to clear non-IList collections (e.g. HashSet<T>) via ICollection<T> before repopulation
Hardened assertions and fixed shadow-property / null-PropertyInfo edge cases
Added unit and integration test coverage for all of the above

### CODE REVIEW

File: src/Couchbase.EntityFrameworkCore/Query/Internal/CouchbaseQuerySqlGenerator.cs [hunk 1]
```diff
@@ -93,6 +103,66 @@ private static bool IsOwnedTable(TableExpression tableExpression)
...
     }
 
+    /// <summary>
+    /// Returns <see langword="true"/> when every table referenced by
+    /// <paramref name="selectExpression"/> (directly and through its own JOINs) is an owned
+    /// type.  Used to detect EF Core's lateral-join subquery pattern for OwnsMany items that
+    /// have nested owned navigations — the whole subquery can be skipped because its data is
+    /// embedded in the parent document.
+    /// </summary>
+    internal static bool IsAllOwnedTablesSelect(SelectExpression selectExpression)
```
Comment (davidkelly) [inline src/Couchbase.EntityFrameworkCore/Query/Internal/CouchbaseQuerySqlGenerator.cs:113]: I believe this only goes 2 levels deep.   I think tables can be nested arbitrarily deep.

**Review by davidkelly — APPROVED** [review]
Ok - the issues I saw are part of a document, they were a bit broader than this PR so they will be handled in a future commit.
